src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+                        registrationService.register(user),
+                "UserRegistrationException expected for empty user.login");
+        assertEquals("User's login is empty", exception.getMessage());
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user = createUser(USER_LOGIN, null, USER_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for user.password = null");
+        assertEquals("User's password is null", exception.getMessage());
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, null);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for user.age = null");
+        assertEquals("User's age is null", exception.getMessage());
+    }
+
+    @Test
+    void register_largeAge_notOk() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, LARGE_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for large user.age");
+        assertEquals("User's age is too big", exception.getMessage());
+    }
+
+    @Test
+    void register_userAlreadyExist_notOk() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, USER_AGE);
+        registrationService.register(user);
+        User newUser = createUser(USER_LOGIN, NEW_USER_PASSWORD, NEW_USER_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(newUser),
+                "UserRegistrationException expected for already stored user");
+        assertEquals("User with such login already exists", exception.getMessage());
+    }
+
+    @Test
+    void register_invalidAge_notOK() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, INVALID_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for invalid user's age");
+        assertEquals("User's age should be at least 18", exception.getMessage());
+    }
+
+    @Test
+    void register_invalidPassword_notOk() {
+        user = createUser(USER_LOGIN, INVALID_PASSWORD, USER_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for invalid user's password");
+        assertEquals("User's password should contain "
+                + "at least 6 characters", exception.getMessage());
+    }
+
+    private User createUser(String login, String password, Integer age) {
```

Comment:
Use BeforeEach annotation for user initialization

Reply:
using before each requires re-using the setter each time to override the parameter

Reply:
Up to you, but parameters which you reset depends on the test, you dont need to rewrite the parameters which you dont want to change

Reply:
if you reuse a class variable, better to create it 1 time and change the fields using setters, then recreate a new user for each test